src/geometry.rs

```diff
@@ -215,16 +215,30 @@ impl Geometry {
         &self.normals
     }
 
+    /// Extracts oriented edges from all mesh faces
     pub fn oriented_edges_iter<'a>(&'a self) -> impl Iterator<Item = OrientedEdge> + 'a {
         self.triangle_faces_iter()
             .flat_map(|face| ArrayVec::from(face.to_oriented_edges()).into_iter())
     }
 
+    /// Extracts unoriented edges from all mesh faces
     pub fn unoriented_edges_iter<'a>(&'a self) -> impl Iterator<Item = UnorientedEdge> + 'a {
         self.triangle_faces_iter()
             .flat_map(|face| ArrayVec::from(face.to_unoriented_edges()).into_iter())
     }
 
+    /// Genus of a mesh is the number of holes in topology / conectivity
+    /// The mesh must be triangular and watertight
+    /// V - E + F = 2 (1 - G)
+    pub fn mesh_genus(&self, edges: &HashSet<UnorientedEdge>) -> i32 {
```

Comment:
I think this came from beyond the grave - there already is a `mesh_genus` function in `mesh_analysis.rs`.